scripts/background.js

```diff
@@ -335,116 +335,114 @@ chrome.runtime.onMessage.addListener(function (message, sender, sendResponse) {
-                      'ocaid' in resp) {
-                      chrome.browserAction.setBadgeText({ tabId: tabId, text: 'B' })
-                    }
-                  })
-              }
-            })
-          }
-          if (event1.auto_update_context === true) {
-            if (event1.show_context === "tab") {
-              if (contexts.findIndex(e => e.tab !== 0) >= 0 || windowIdtest !== 0) {
-                chrome.tabs.query({
-                  windowId: windowIdtest
-                }, function (tabs) {
-                  var tab1 = tabs[0];
-                  tabIdtest = tab1.id;
-                  if (tab.id !== tabIdtest && contexts.findIndex(e => e.tab !== tab.id) >= 0 && tabs.filter(e => e.id === tabId).length === 0 && contexts.filter(e => e.tab === tabId).length === 0 && contexts.filter(e => e.tab === tab.id).length === 0) {
-                    for (var i = 0; i < contexts.length; i++) {
-                      var e = contexts[i];
-                      chrome.tabs.update(parseInt(e.tab), { url: e.htmlUrl + urlsToAppend[i] });
-                    };
+    var received_url = tab.url;
+    if (!(received_url.includes("chrome://newtab/") || received_url.includes("chrome-extension://") || received_url.includes("alexa.com") || received_url.includes("whois.com") || received_url.includes("twitter.com") || received_url.includes("oauth"))) {
+      singlewindowurl = received_url;
+      tagcloudurl = new URL(singlewindowurl);
+      received_url = received_url.replace(/^https?:\/\//, '');
+      var last_index = received_url.indexOf('/');
+      var url = received_url.slice(0, last_index);
+      var open_url = received_url;
+      if (open_url.slice(-1) === '/') {
+        open_url = received_url.substring(0, open_url.length - 1);
+      }
+      var urlsToAppend = [url, tab.url, open_url, tab.url, tab.url, tagcloudurl];
+      chrome.storage.sync.get(['books', 'auto_update_context', 'show_context'], function (event1) {
+        if (event1.books === true) {
+          chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
+            url = tabs[0].url;
+            tabId = tabs[0].id;
+            if (url.includes('www.amazon')) {
+              fetch('https://archive.org/services/context/amazonbooks?url=' + url)
+                .then(resp => resp.json())
+                .then(resp => {
+                  if (('metadata' in resp && 'identifier' in resp['metadata']) ||
+                    'ocaid' in resp) {
+                    chrome.browserAction.setBadgeText({ tabId: tabId, text: 'B' })
                   }
-                });
-              }
-            } else if (event1.show_context === "singlewindow") {
+                })
+            }
+          })
+        }
+        if (event1.auto_update_context === true) {
+          if (event1.show_context === "tab") {
+            if (contexts.findIndex(e => e.tab !== 0) >= 0 || windowIdtest !== 0) {
```

Comment:
Inside `if (event1.show_context === "tab") {` there is only one `if` and nothing else.
Can you make it 
```
if (event1.show_context === "tab" && (contexts.findIndex(e => e.tab !== 0) >= 0 || windowIdtest !== 0) ) {
```
to simplify even more?